calculator_test.go

```diff
@@ -2,23 +2,293 @@ package calculator_test
 
 import (
 	"calculator"
+	"math"
+	"math/rand"
 	"testing"
+	"time"
 )
 
-func TestAdd(t *testing.T) {
+func closeEnough(a, b, epsilon float64) bool {
+	if math.Abs(a-b) < epsilon {
+		return true
+	}
+	return false
+}
+
+const testcount = 100
+const testcaseMaxsize = 10
+
+func TestAddRand(t *testing.T) {
+	t.Parallel()
+	seed := rand.NewSource(time.Now().UnixNano())
+	r := rand.New(seed)
+	count := 0
+	// run the test addcount times
+	for count < testcount {
+		// get a random size for the number of values
+		// to add together
+		nums := make([]float64, r.Intn(testcaseMaxsize))
+		// initialize nums slice to contain random values
+
+		arg1 := r.NormFloat64()
+		arg2 := r.NormFloat64()
+
+		want := arg1 + arg2
+		for i := range nums {
+			nums[i] = r.NormFloat64()
+			want += nums[i]
+		}
+		var testCases = []struct {
+			name string
+			arg1 float64
+			arg2 float64
+			nums []float64
+			want float64
+		}{
+			{name: "Random value data", arg1: arg1, arg2: arg2, nums: nums, want: want},
```

Comment:
This is a good example of where our test case struct is a little different; we don't need a name for each random case! We don't need `want` either, since we can just calculate it inside the test loop.

Reply:
Actually, I think we don't need the test case anonymous struct at all.